modules/administration_manual/pages/maintenance/encryption/migrating-from-user-key-to-master-key.adoc

```diff
@@ -0,0 +1,128 @@
+
+The decryption workflow described here works for both MySQL and SQLite based databases. It will only work when users have enabled the password recovery. Moreover, an admin recovery password needs to be activated and available for the ownCloud administrator.
+
+If you need to migrate from User Key-based to Master Key-based encryption, there are several steps that you need to follow to ensure a smooth and complete transition:
+
+. xref:disable-user-key-based-encryption[Disable User Key-based encryption]
+. xref:remove-the-encryption-records-from-the-owncloud-database[Remove the encryption records from the ownCloud database]
+. xref:remove-the-files_encryption-directory[Remove the `files_encryption` directory]
+. xref:encrypt-the-filesystem-using-master-key-based-encryption[Encrypt the filesystem using Master Key-based encryption]
+
+
+[[disable-user-key-based-encryption]]
+== Disable User Key-based Encryption
+
+The first part of the migration process is to decrypt all files and to disable encryption in ownCloud, which requires three commands to be executed.
+These commands are: xref:configuration/server/occ_command.adoc#encryption[`occ encryption:decrypt-all` and `occ encryption:disable`], and xref::configuration/server/occ_command.adoc#apps-commands[`occ app:disable`].
+
+You can see an example of calling the commands listed below, configured to require no user interaction.
+
+[source,console]
+----
+sudo -u www-data php occ encryption:decrypt-all --continue=yes && \
+  sudo -u www-data php occ encryption:disable --no-interaction && \
+  sudo -u www-data php occ app:disable --no-interaction encryption
+----
+
+NOTE: The decryption of the files by the ownCloud system administrator requires the current passwords of all users!
+
+[[remove-the-encryption-records-from-the-owncloud-database]]
+== Remove the Encryption Records from the ownCloud Database
+
+Once your ownCloud files are unencrypted, and encryption has been disabled, you need to remove the encryption records from the database.
+There is, currently, no `occ` command to handle this, so it has to be done manually.
+Specifically, you need to remove all records from the `oc_appconfig` table where the `appid` column is set to `encryption`.
+
+In the examples below you can see how to do this using the SQLite database.
+If you are not using SQLite, please use the commands specific to your database vendor.
+
+[NOTE]
+====
+The example code assumes that the path to the SQLite database is `<YOUR/OWNCLOUD/ROOT/DIRECTORY>data/owncloud.database`.
+====
+
+[source,console]
+----
+sqlite3 data/owncloud.database
```

Comment:
I do not think it is a good idea describing a migration path with sqlite. Use mysql (with proper commands) instead. sqlite is, as described, discuraged for production, therefore using mysql is the choice.

Reply:
@mmattel I only added the mysql parts, the sqlite stuff was there before. But you are right, it would be better to remove sqlite in the example. 

on forking: I work from a fork (https://github.com/thommierother/docs) with atom as the editor in the background. Sorry, if something went wrong. But we can also delete the doc which is on the upstream master and re-add  it from my fork. If that makes sense, I don't know ....

@settermjd : what do you think?

Reply:
If I look into the actual docs, there is no mentioning of sqlite. Pls the a look on docs.owncloud.org There must be a mismatch big mismatch between your fork and origin/master...

Reply:
lets wait for @settermjd. The original page was already there and it contained the sqlite stuff (I never used sqlite for OC myself). He also reviewed when I was working on it for the first time (end of 18 or beginning of 19 I think).  I just cloned the current docs repo and did not change anything in the file structure of the repo. The only doc i am working on is the encryption one. But if we have any file/structure conflicts, I can re-create my fork repo from the master and we can start to add the encryption doc again, no problem from my side.

Reply:
> end of 18 or beginning of 19 I think

You mean 2018 or 2019?
In that case your fork urgently needs an update...